### PR TITLE
Add Cats activities, comparison and tests

### PR DESCRIPTION
**Pull request by DenisLobur**

### CODE REVIEW

File: src/main/java/lesson10/Predator.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package lesson10;
+
+public abstract class Predator {
+
+    protected Color color;
```
Comment (nikmikhailov13) [inline src/main/java/lesson10/Predator.java]: why not private?

File: src/test/java/lesson10/CatTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        assertEquals(bob, tod);
+        assertNotEquals(bob, pod);
+        assertNotEquals(tod, pod);
+        // checking if cat.equals(cat)
+        assertEquals(bob, bob);
+        // checking if cat.equals(obj)
+        assertNotEquals(bob, new Object());
+    }
+
+    @ParameterizedTest
```
Comment (nikmikhailov13) [inline src/test/java/lesson10/CatTest.java:35]: 👍

**Review by nikmikhailov13 — APPROVED** [review]
LGTM

File: src/main/java/lesson10/Predator.java (hunk 1 above)
Comment (DenisLobur) [reply src/main/java/lesson10/Predator.java]: yeap, my bad. Fixed

**Pull request merged**